File: backend/app/schemas/weather.py
from pydantic import BaseModel
from typing import List, Optional

class Coordinates(BaseModel):
    latitude: float
    longitude: float

class CurrentWeather(BaseModel):
    temperature: float
    apparent_temperature: float
    humidity: int
    rainfall: float
    wind_speed: float
    weather_code: int
    weather_condition: str

class ForecastItem(BaseModel):
    date: str
    day_index: int
    temp_max: Optional[float] = None
    temp_min: Optional[float] = None
    weather_code: int
    weather_condition: str
    precipitation_sum: float
    precipitation_probability: int

class WeatherResponse(BaseModel):
    success: bool
    location: Optional[str] = None
    resolved_location: Optional[str] = None
    coordinates: Optional[Coordinates] = None
    current: Optional[CurrentWeather] = None
    forecast: Optional[List[ForecastItem]] = None
    farming_insights: Optional[List[str]] = None
    irrigation_advisory: Optional[str] = None
    updated_at: Optional[str] = None
    error_type: Optional[str] = None
    message: Optional[str] = None

File: frontend/src/types/weather.ts
export interface Coordinates {
  latitude: number;
  longitude: number;
}

export interface CurrentWeather {
  temperature: number;
  apparent_temperature: number;
  humidity: number;
  rainfall: number;
  wind_speed: number;
  weather_code: number;
  weather_condition: string;
}

export interface ForecastItem {
  date: string;
  day_index: number;
  temp_max?: number | null;
  temp_min?: number | null;
  weather_code: number;
  weather_condition: string;
  precipitation_sum: number;
  precipitation_probability: number;
}

export interface WeatherResponse {
  success: boolean;
  location?: string | null;
  resolved_location?: string | null;
  coordinates?: Coordinates | null;
  current?: CurrentWeather | null;
  forecast?: ForecastItem[] | null;
  farming_insights?: string[] | null;
  irrigation_advisory?: string | null;
  updated_at?: string | null;
  error_type?: string | null;
  message?: string | null;
}
